junebug/tests/test_api.py

```diff
@@ -57,14 +57,26 @@ def test_http_error(self):
 
     @inlineCallbacks
     def test_get_channel_list(self):
+        redis = yield self.get_redis()
+        config = self.default_channel_config
+
         resp = yield self.get('/channels')
-        yield self.assert_response(
-            resp, http.INTERNAL_SERVER_ERROR, 'generic error', {
-                'errors': [{
-                    'message': '',
-                    'type': 'NotImplementedError',
-                    }]
-                })
+        yield self.assert_response(resp, http.OK, 'channels listed', [])
+
+        yield Channel(redis, {}, config, u'test-channel-1').save()
+
+        resp = yield self.get('/channels')
+        yield self.assert_response(resp, http.OK, 'channels listed', [
+            u'test-channel-1',
```

Comment:
Hmm. Names would be nice, also had that thought when I started on this PR. UUIDS are pretty useful by themselves though, they can serve as a starting point to find out more information about each channel via further api calls. I think its common for `GET`s to a collection URI to return a list of URIs of its members, rather than ids or the full object for each member (sounds like it could get pretty large pretty quickly, but maybe not). Returning just a few fields, like the name and id seems a bit weird to me, get the feeling we'd eventually end up returning the entire object for each member at some point if we did that.

That all said, I have a slight preference for UUIDS or URIs, but happy to go with returning the full objects if people prefer.